lib/matplotlib/quiver.py

```diff
@@ -329,18 +313,16 @@ def on_dpi_change(fig):
 
         if self.labelcolor is not None:
             self.text.set_color(self.labelcolor)
-        self._initialized = False
+        self._dpi_at_last_init = None
         self.zorder = Q.zorder + 0.1
 
-    def remove(self):
-        # docstring inherited
-        self.Q.axes.figure.callbacks.disconnect(self._cid)
-        self._cid = None
-        super().remove()  # pass the remove call up the stack
+    @property
+    def labelsep(self):
+        return self._labelsep_inches * self.Q.axes.figure.dpi
 
     def _init(self):
-        if True:  # not self._initialized:
-            if not self.Q._initialized:
+        if True:  # self._dpi_at_last_init != self.axes.figure.dpi
```

Comment:
I don't understand why this condition is here if its always true?

Reply:
As above.